delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala

```diff
@@ -496,19 +509,46 @@ object Storages {
     )
   }
 
-  private[storage] def next(
+  private def encryptSource(source: DecryptedSecret[Json], value: StorageValue[Encrypted]): EncryptedSecret[Json] =
+    source.mapEncrypt(changeSourceSensitiveValues(value, _))
+
+  private def decryptSource(source: EncryptedSecret[Json], value: StorageValue[Decrypted]): DecryptedSecret[Json] =
+    source.mapDecrypt(changeSourceSensitiveValues(value, _))
+
+  private def changeSourceSensitiveValues[A <: EncryptionState](value: StorageValue[A], source: Json): Json =
```

Comment:
We could just encrypt the whole source payload to make this simpler and not changing the incoming payload

Reply:
We don't want this, because it will make it less readable for debugging purposes. You still want to be able to read easily the rest of the fields

Reply:
We have StorageValue for debugging purposes

Reply:
But I want to debug what the client posted, not what is in the value (then I don't know if the client posted it or not)

Reply:
Besides, in 99% of the cases no client is going to provide credentials or accessKey/secretKey through the API. 
Instead they are going to rely on the configuration values, which makes the encryption of the whole json an overkill.